Source/Modules/python.cxx

```diff
@@ -4345,6 +4352,133 @@ class PYTHON:public Language {
+    printSlot2(f, getSlot(n, "feature:python:nb_float"), "nb_float", "unaryfunc");
+    printSlot2(f, getSlot(n, "feature:python:nb_inplace_add"), "nb_inplace_add", "binaryfunc");
+    printSlot2(f, getSlot(n, "feature:python:nb_inplace_subtract"), "nb_inplace_subtract", "binaryfunc");
+    printSlot2(f, getSlot(n, "feature:python:nb_inplace_multiply"), "nb_inplace_multiply", "binaryfunc");
+    printSlot2(f, getSlot(n, "feature:python:nb_inplace_remainder"), "nb_inplace_remainder", "binaryfunc");
+    printSlot2(f, getSlot(n, "feature:python:nb_inplace_power"), "nb_inplace_power", "ternaryfunc");
+    printSlot2(f, getSlot(n, "feature:python:nb_inplace_lshift"), "nb_inplace_lshift", "binaryfunc");
+    printSlot2(f, getSlot(n, "feature:python:nb_inplace_rshift"), "nb_inplace_rshift", "binaryfunc");
+    printSlot2(f, getSlot(n, "feature:python:nb_inplace_and"), "nb_inplace_and", "binaryfunc");
+    printSlot2(f, getSlot(n, "feature:python:nb_inplace_xor"), "nb_inplace_xor", "binaryfunc");
+    printSlot2(f, getSlot(n, "feature:python:nb_inplace_or"), "nb_inplace_or", "binaryfunc");
+    printSlot2(f, getSlot(n, "feature:python:nb_floor_divide"), "nb_floor_divide", "binaryfunc");
+    printSlot2(f, getSlot(n, "feature:python:nb_divide"), "nb_true_divide", "binaryfunc");
+    printSlot2(f, getSlot(n, "feature:python:nb_inplace_floor_divide"), "nb_inplace_floor_divide", "binaryfunc");
+    printSlot2(f, getSlot(n, "feature:python:nb_inplace_divide"), "nb_inplace_true_divide", "binaryfunc");
+    printSlot2(f, getSlot(n, "feature:python:nb_index"), "nb_index", "unaryfunc");
+
+    // sequence
+    printSlot2(f, getSlot(n, "feature:python:sq_length"), "sq_length", "lenfunc");
+    printSlot2(f, getSlot(n, "feature:python:sq_concat"), "sq_concat", "binaryfunc");
+    printSlot2(f, getSlot(n, "feature:python:sq_repeat"), "sq_repeat", "ssizeargfunc");
+    printSlot2(f, getSlot(n, "feature:python:sq_item"), "sq_item", "ssizeargfunc");
+    printSlot2(f, getSlot(n, "feature:python:sq_ass_item"), "sq_ass_item", "ssizeobjargproc");
+    printSlot2(f, getSlot(n, "feature:python:sq_contains"), "sq_contains", "objobjproc");
+    printSlot2(f, getSlot(n, "feature:python:sq_inplace_concat"), "sq_inplace_concat", "binaryfunc");
+    printSlot2(f, getSlot(n, "feature:python:sq_inplace_repeat"), "sq_inplace_repeat", "ssizeargfunc");
+
+    Printf(f, "  {Py_tp_members, members},\n");
+    Printf(f, "  { 0, NULL}\n");
+    Printf(f, "  };\n");
+    Printf(f, "PyType_Spec spec = {\n");
+    Printf(f, "  %s,\n", quoted_symname);
+    Printf(f, "  sizeof(SwigPyObject),\n");
+    Printf(f, "  0,\n");
+    Printf(f, "  %s,\n", getSlot(n, "feature:python:tp_flags", tp_flags_py3), "tp_flags");
+    Printf(f, "  slots\n");
+    Printf(f, "  };\n");
+    Printv(f, "  PyObject* tuple_bases = SwigPyBuiltin_InitBases(bases);\n", NIL);
+    Printf(f, "  PyTypeObject* pytype = (PyTypeObject*)PyType_FromSpecWithBases(&spec, tuple_bases);\n");
```

Comment:
And here I'm also not sure about the expected lifetime of the new type.
Does `SWIGINTERN` mean it's only created once per process?

Reply:
It's a macro which is essentially defined to `static` (possibly plus compiler-specific stuff to suppress unused warnings) if not already defined.  There seems to be no documentation of why it exists beyond `/* internal SWIG method */` and the VCS history doesn't help, but I guess it's useful to be able to override it in some situation.  Or maybe someone just saw the "define if not already defined" pattern for existing macros which are useful to override and copied it for one which isn't.

So you'll get one per SWIG-wrapped module loaded rather than one per process (unless you pre-defined `SWIGINTERN` to something to change that).

Reply:
The lifetime of this type depends on the users script.  Anytime an c object is constructed in python, this type is used.